### DISCUSSION TITLE
Withdrawing - Logic with .send / require statment

### DISCUSSION THREAD

**Discussion by sebastianahimsa** [Cyfrin/foundry-full-course-cu#251] [Q&A] [answered]
When it comes the part of sending... i don't understand the logic of the line:
bool sendSuccess = payable(msg.sender).send(address(this).balance); ((FROM WHERE IT COMES THIS  ".balance" ?? ))
require(sendSuccess, "Send failed");

this last part: the one(sendSuccess) or the other("Send failed")) will be selected depending the Bool line?
If the bool send false the require will chose the second statment ("Send failed") ?

**Comment by ghost** [comment, marked as answer]
The `send` method returns a boolean to indicate if the transactions was successfull or not. The `require` function is used to verify a condition in this case it will verify if `sendSuccess` is true, if it not, it will revert the transaction with the message you provided.

Now, it not recommended to use `send` to transfer ether, currently the recommended method is `call`. For reference: https://solidity-by-example.org/sending-ether/.

**Reply by sebastianahimsa** [reply]
Perfect. Thank u very much!

**Comment by pyromaus** [comment]
```solidity
bool sendSuccess = payable(msg.sender).send(address(this).balance); 
require(sendSuccess, "Send failed");
```
in Solidity calling `.balance` on any address tells you its balance in Wei 
`1 ether` = 1 000 000 000 000 000 000 Wei 
```solidity
address private sebastian;
uint256 lunchMoney = sebastian.balance;
address public burgerKing;

bool hungerSatisfied = payable(burgerKing).send(lunchMoney);
require(hungerSatisfied, "Damn, they're closed");
```

**Reply by sebastianahimsa** [reply]
very clear, thanks !!
